src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/EvaluationContext.cs

```diff
@@ -517,6 +530,86 @@ private static MethodSymbol GetSynthesizedMethod(CommonPEModuleBuilder moduleBui
             }
         }
 
+        private static void GetAliasLocals(
+            ArrayBuilder<LocalSymbol> builder,
+            ImmutableArray<Alias> aliases,
+            MethodSymbol method,
+            CSharpCompilation compilation)
+        {
+            var sourceAssembly = compilation.SourceAssembly;
+            var typeNameDecoder = new EETypeNameDecoder(compilation, (PEModuleSymbol)method.ContainingModule);
+            foreach (var alias in aliases)
+            {
+                LocalSymbol additionalLocal;
+                builder.Add(CreateAliasLocal(typeNameDecoder, method, sourceAssembly, alias, out additionalLocal));
+                if ((object)additionalLocal != null)
+                {
+                    Debug.Assert(alias.Kind == DkmClrAliasKind.ReturnValue);
+                    builder.Add(additionalLocal);
+                }
+            }
+        }
+
+        private static LocalSymbol CreateAliasLocal(
+            TypeNameDecoder<PEModuleSymbol, TypeSymbol> typeNameDecoder,
+            MethodSymbol containingMethod,
+            AssemblySymbol sourceAssembly,
+            Alias alias,
+            out LocalSymbol additionalLocal)
```

Comment:
If this method can generate multiple locals, consider passing in the ArrayBuilder instead.